pkg/pillar/types/zedmanagertypes.go

```diff
@@ -198,6 +198,17 @@ func (statusPtr *AppInstanceStatus) ClearError() { //revive:disable-line
 	statusPtr.ErrorTime = time.Time{}
 }
 
+// HasUnderlayMacAddr - Check if the AI status has the underlay network with this Mac Address
+func (status *AppInstanceStatus) HasUnderlayMacAddr(MacAddr, ipAddr string) bool {
+	for idx, ulStatus := range status.UnderlayNetworks {
+		if ulStatus.VifInfo.Mac == MacAddr {
+			status.UnderlayNetworks[idx].AllocatedIPAddr = ipAddr
```

Comment:
The function name reads as a predicate, which makes it confusing to have it also update the status.
Should it be named MaybeUpdateIPAddr() instead, or make the caller update the status?

Reply:
Sure.